integration/users/connection_test.go

```diff
@@ -238,3 +238,167 @@ func TestAuthentication(t *testing.T) {
 		})
 	}
 }
+
+// TestAuthenticationEnableNewAuthNoUser tests that the backend authentication
+// is used when there is no user in the database. This ensures that there is
+// some form of authentication even if there is no user.
+func TestAuthenticationEnableNewAuthNoUserExists(t *testing.T) {
+	t.Parallel()
+
+	s := setup.SetupWithOpts(t, nil)
+	ctx := s.Ctx
+	collection := s.Collection
+	db := collection.Database()
+
+	// drop the user created in the setup
+	err := db.Client().Database("admin").RunCommand(ctx, bson.D{
+		{"dropUser", "username"},
+	}).Err()
+	require.NoError(t, err, "cannot drop user")
+
+	testCases := map[string]struct {
+		username  string
+		password  string
+		mechanism string
+
+		pingErr   string
+		insertErr string
+	}{
+		"PLAINNonExistingUser": {
+			username:  "plain-user",
+			password:  "whatever",
+			mechanism: "PLAIN",
+			insertErr: `role "plain-user" does not exist`,
+		},
+		"PLAINBackendUser": {
+			username:  "username",
+			password:  "password",
+			mechanism: "PLAIN",
+		},
+		"SHA256NonExistingUser": {
+			username:  "sha256-user",
+			password:  "whatever",
+			mechanism: "SCRAM-SHA-256",
+			pingErr:   "Authentication failed",
+		},
+	}
+
+	for name, tc := range testCases {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			if tc.mechanism == "PLAIN" {
+				setup.SkipForMongoDB(t, "PLAIN mechanism is not supported by MongoDB")
+			}
+
+			t.Parallel()
+
+			credential := options.Credential{
+				AuthMechanism: tc.mechanism,
+				AuthSource:    db.Name(),
```

Comment:
This is also a param I don't understand :) Shouldn't it be `admin` instead of the current DB name?
I see parsing this param is not implemented on the FerretDB side yet, and I don't see an issue for it... Maybe we need to clarify how we expect it to work.